### PR TITLE
Setup Travis CI with lint checks

### PR DESCRIPTION
**Pull request by ananyo2012**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+  - "6"
```
Comment (ryzokuken) [inline .travis.yml]: Why don't we bump this to v7 atleast? v8 will soon be LTS and v7 brings some important upgrades like `async` and `await`.

Comment (jywarren) [inline .travis.yml]: Try it out -- open a new PR and see how it does! But can't you also do both? https://github.com/publiclab/PublicLab.Editor/blob/master/.travis.yml#L3-L4

**Comment by jywarren** [comment]
Is yarn.lock necessary - like Gemfile.lock on plots2?

**Comment by ryzokuken** [comment]
@jywarren https://yarnpkg.com/lang/en/docs/yarn-lock/

File: .travis.yml (hunk 1 above)
Comment (ananyo2012) [inline .travis.yml]: Ok then we can test against both v6 and v7. Does that sound right?

Comment (ryzokuken) [inline .travis.yml]: I see Travis picked 6 up from `.travis.yml` and did `nvm install 6`. Maybe we could make it do `nvm install node` (like in the README, gets the latest version) instead? 

BTW, for the time being atleast, v6 and v7 sound good, although that would double the time it takes, so we should really think about it. If we decide to go with two versions, we can switch to v7 and v8 later.

Comment (ryzokuken) [inline .travis.yml]: @ananyo2012 Looking at the docs, I realized that it is infact possible to use the latest version and not a locked to a specific version. https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Specifying-Node.js-versions for more. Thanks.

**Comment by ryzokuken** [comment]
@ananyo2012 Apparently, this needs to be merged before anything else.

File: .travis.yml (hunk 1 above)
Comment (ananyo2012) [inline .travis.yml]: Well the purpose of using multiple versions is ensuring we support the app for multiple versions of node. NOT from the point of view how fast the test runs. Also v6 is the current LTS version so I thought of using it.

**Comment by ananyo2012** [comment]
Yup I am going to merge this one right now!

**Comment by ananyo2012** [comment]
@jywarren Including yarn.lock also helps to cache dependencies in Travis.

**Pull request merged**